source/OpenCDMSession.cpp

```diff
@@ -264,6 +264,48 @@ void OpenCDMSession::addProtectionMeta(GstBuffer *buffer, GstBuffer *subSample,
     rialto_mse_add_protection_metadata(buffer, info);
 }
 
+bool OpenCDMSession::addProtectionMeta(GstBuffer *buffer, GstCaps *caps)
+{
+    if (!caps)
+    {
+        TRACE_L1("Invalid caps");
+        return false;
+    }
+
+    GstStructure *structure = gst_caps_get_structure(caps, 0);
```

Comment:
From reading the description of the new API here https://github.com/rdkcentral/ThunderClientLibraries/blob/master/Source/ocdm/adapter/open_cdm_adapter.h, I think it's saying the the structure will be attached to 'buffer' as metadata already? Does this mean our opencdm_gstreamer_session_decrypt_buffer() can just be a stub and rialto-gstreamer can extract the data directly?

Reply:
Yes I think you are write, the metadata will be added to the buffer already, however, current implementation of cobalt deletes the protection meta after calling decrypt so we need to add our new, rialto_protection_meta to the buffer so that it persists. I suspect this is going to be the same for the new implementation, so I have copied the structure and added to the buffer as a rialto_protection_meta as before.